include/cuco/detail/static_map.inl

```diff
@@ -133,29 +133,57 @@ __device__ bool static_map<Key, Value, Scope, Allocator>::device_mutable_view::i
     using cuda::std::memory_order_relaxed;
     auto expected_key   = this->get_empty_key_sentinel();
     auto expected_value = this->get_empty_value_sentinel();
-    auto& slot_key      = current_slot->first;
-    auto& slot_value    = current_slot->second;
-
-    bool key_success =
-      slot_key.compare_exchange_strong(expected_key, insert_pair.first, memory_order_relaxed);
-    bool value_success =
-      slot_value.compare_exchange_strong(expected_value, insert_pair.second, memory_order_relaxed);
-
-    if (key_success) {
-      while (not value_success) {
-        value_success =
-          slot_value.compare_exchange_strong(expected_value = this->get_empty_value_sentinel(),
-                                             insert_pair.second,
-                                             memory_order_relaxed);
+
+    key_type const existing_key = current_slot->first.load(memory_order_relaxed);
+    // the key we are trying to insert is already in the map, so we return with failure to insert
+    if (key_equal(existing_key, insert_pair.first)) { return false; }
+
+    // Packed CAS for 4B/4B (or less) key/value pairs
+    if constexpr (sizeof(Key) <= 4 and sizeof(Value) <= 4) {
```

Comment:
This won't work if Key and Value are 2B types. You should take a look at the `is_packable` logic in cudf.

Reply:
That's what I was gonna fix.